keras/backend/tensorflow_backend.py

```diff
@@ -1202,8 +1202,11 @@ def rnn(step_function, inputs, initial_states,
     if mask is not None:
         if mask.dtype != tf.bool:
             mask = tf.cast(mask, tf.bool)
-        if len(mask.get_shape()) == ndim - 1:
-            mask = expand_dims(mask)
+        if len(mask.get_shape()) < ndim:
+            tmp = ndim - len(mask.get_shape())
+            for _ in range(tmp):
+                mask = expand_dims(mask)
+        assert len(mask.get_shape()) == ndim
```

Comment:
Thanks for your advice. We add a ValueError as follows:

        if len(mask.get_shape()) > ndim:
            raise ValueError('The mask should not have a higher rank than x. '
                             'Found rank of mask is ' + str(len(mask.get_shape())) + ' rank of x is ' + str(ndim))